pkg/registry/common/querycache/nse_client.go

```diff
@@ -64,7 +64,7 @@ func (q *queryCacheNSEClient) Find(ctx context.Context, query *registry.NetworkS
 
 	resultCh := make(chan *registry.NetworkServiceEndpoint, len(nses))
 	for _, nse := range nses {
-		resultCh <- nse
+		resultCh <- nse.Clone()
 		q.storeInCache(ctx, nse, opts...)
```

Comment:
Can we also store the clone in the cache to prevent possible issues?
```suggestion
                nseClone := nse.Clone()
		resultCh <- nseClone
		q.storeInCache(ctx, nseClone, opts...)
```

Reply:
There is no actual need for this, because we are already receiving cloned entities from `Find`. So the problem is that `querycache` violates this assumption returning not cloned entity from `Find`.

Reply:
Will localbypass then modify the stored value in the cache?

Reply:
In existing code / with your suggestion - yes.
In such case: `resultCh <- nse.Clone()` - no.

Reply:
ok, as I understand we still have the problem if we'll move local bypass after querycache. Is it correct?
Suggestion updated.

Reply:
Yep, the problem with data race will be revealed somewhere else in the future.